src/pipeline_v3/utils/config.py

```
@@ -114,6 +114,37 @@ class SearchSettings:
     default_limit: int = 5  # Default number of search results
     default_mode: str = "hybrid"  # Default search mode: hybrid, vector, or keyword
 
+@dataclass
+class ProcessingProfile:
+    """Defines a reusable processing configuration profile."""
+    document_type: str = "auto"
+    processing_options: List[str] = field(default_factory=list)
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)